src/ts/sign.ts

```diff
@@ -0,0 +1,120 @@
+import { BytesLike, ethers, Signer } from "ethers";
+
+import { hashOrder, Order, ORDER_TYPE_FIELDS, timestamp } from "./order";
+import { isTypedDataSigner, TypedDataDomain } from "./types/ethers";
+
+/**
+ * The signing scheme used to sign the order.
+ */
+export const enum SigningScheme {
+  /**
+   * The EIP-712 typed data signing scheme. This is the preferred scheme as it
+   * provides more infomation to wallets performing the signature on the data
+   * being signed.
+   *
+   * <https://github.com/ethereum/EIPs/blob/master/EIPS/eip-712.md#definition-of-domainseparator>
+   */
+  ERC712,
```

Comment:
did you mean
```suggestion
  EIP712,
```

Reply:
I thought it was a standard, but you are right, it isn't. Probably I got confused with `ERC721`.
This means that two of the three schemes we use are potentially going to change in the future.

Reply:
🙈